src/services/allocation.js

```diff
@@ -1,8 +1,56 @@
+import { parseFilters } from "../util";
 import client from "./api_client";
 
+// Flag to enable mock data - must be explicitly set via REACT_APP_USE_MOCK_DATA environment variable
+const USE_MOCK_DATA = process.env.REACT_APP_USE_MOCK_DATA === "true";
+
+// Mock data for testing drilldown functionality when backend is not available
+// Only used when REACT_APP_USE_MOCK_DATA=true is explicitly set
+const getMockData = (aggregate, filters) => {
```

Comment:
Nit: Can this function go in its own file alongside this one, so we keep test / mock utilities separate from core service logic?